src/types/shared.ts

```diff
@@ -112,28 +147,7 @@ export type FlightsConditions = {
    * In this case you should refer to the slices conditions to determine if any part of the order or offer is changeable.
    * If the airline hasn't provided any information about whether this order or offer can be changed then this property will be `null`.
    */
-  change_before_departure?: {
-    /**
-     * The currency of the `penalty_amount` as an [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217) currency code.
-     * This will be in a currency determined by the airline, which is not necessarily the same as the currency of the order or offer.
-     * If this is `null` then `penalty_amount` will also be `null`.
-     * @example "GBP"
-     */
-    penalty_currency?: string | null
-    /**
-     * If the modification is `allowed` then this is the amount payable to apply the modification to all passengers.
-     * If there is no penalty, the value will be zero. If the modification isn't `allowed` or the penalty is not known then this field will be `null`.
-     * If this is `null` then the `penalty_currency` will also be null.
-     * @example "100.00"
-     */
-    penalty_amount?: string | null
-    /**
-     * Whether this kind of modification is allowed post-booking
-     *
-     * @example "true"
-     */
-    allowed: boolean
-  } | null
+  change_before_departure?: FlightsCondition | null
```

Comment:
I don't think we should allow `undefined` in the types, they might mismatch at times if the only expectation is `Something` or `null`.
  
```suggestion
  change_before_departure: FlightsCondition | null
```